macOS/DuckDuckGo/PinnedTabs/Model/PinnedTabsManagerProvider.swift

```diff
@@ -123,21 +124,22 @@ final class PinnedTabsManagerProvider: @preconcurrency PinnedTabsManagerProvidin
     // MARK: Providing PinnedTabsManagers
 
     @MainActor
-    func getNewPinnedTabsManager(shouldMigrate: Bool = false, tabCollectionViewModel: TabCollectionViewModel) -> PinnedTabsManager {
+    func getNewPinnedTabsManager(shouldMigrate: Bool = false, tabCollectionViewModel: TabCollectionViewModel, forceActive: Bool? = nil) -> PinnedTabsManager {
         switch pinnedTabsMode {
         case .separate:
             return getNewPerWindowPinnedTabsManager(shouldMigrate: shouldMigrate,
-                                                    tabCollectionViewModel: tabCollectionViewModel)
+                                                    tabCollectionViewModel: tabCollectionViewModel,
+                                                    forceActive: forceActive)
         case .shared:
             return getSharedPinnedTabManager(shouldMigrate: shouldMigrate)
         }
     }
 
     @MainActor
-    private func getNewPerWindowPinnedTabsManager(shouldMigrate: Bool, tabCollectionViewModel: TabCollectionViewModel) -> PinnedTabsManager {
+    private func getNewPerWindowPinnedTabsManager(shouldMigrate: Bool, tabCollectionViewModel: TabCollectionViewModel, forceActive: Bool? = nil) -> PinnedTabsManager {
         let newPinnedTabsManager = PinnedTabsManager()
         let isFirstWindow = windowControllerManager.mainWindowControllers.isEmpty
-        let isActiveWindow = windowControllerManager.lastKeyMainWindowController?.mainViewController.tabCollectionViewModel === tabCollectionViewModel
+        var isActiveWindow: Bool = forceActive ?? (windowControllerManager.lastKeyMainWindowController?.mainViewController.tabCollectionViewModel === tabCollectionViewModel)
```

Comment:
The test was failing occasionally because this property wasn't true for any window. This isn't possible in real scenarios, so I decided to just force the flag in tests.